### DISCUSSION TITLE
Why aren't we using the built-in Logic Apps expressions `json` & `xml` instead exposing our own XML-JSON component?

### DISCUSSION THREAD

**Discussion by stijnmoreels** [invictus-integration/docs-ifa#237] [Q&A]
The Logic Apps expressions include [`json`](https://learn.microsoft.com/en-us/azure/logic-apps/workflow-definition-language-functions-reference#json) and [`xml`](https://learn.microsoft.com/en-us/azure/logic-apps/workflow-definition-language-functions-reference#xml) to transform between the two, without the need for external calls or Azure Integration Accounts. It was specifically designed to be built-in, fast, and independent.
Why would we then expose such XML-JSON component ourselves?

**Comment by pim-simons** [comment]
I personally have not used these components, but did find information in [this thread](https://teams.microsoft.com/l/message/19:94dc05d412d84c16858094ff79af241d@thread.skype/1592911850709?tenantId=7517bc42-bcf8-4916-a677-b5753051f846&groupId=f2ebbeb0-4e8b-4764-9835-98011ae154e9&parentMessageId=1592911850709&teamName=Codit%20Software&channelName=Invictus%20for%20Azure&createdTime=1592911850709) which also includes the initial design.

Your question is not new, see:

> how is this component different than the standard json() and xml() statements available in Logic Apps?

> At one of my customers we map json to xml by using the xml() statement and using that output to call the transco component. What does this component offer as extra functionality?

> well, does this component still makes sense if you can map json to xml by using the xml() statement directly on Logic Apps?

What this component apparently adds is to be able to also execute a mapping, which I still think is a strange place to do so (we should use the transco for that) but that was my comment in that thread back than also. 

As you said, I just use the built in Logic App expressions for this, no need for an extra call to our framework. I do see some value in these components if you would need to move from JSON to XML or vice-versa outside of Logic Apps maybe.
@GoutsmitSam @Pauwelz we should checkout which projects this component is used and what the usecase for it is.

**Reply by stijnmoreels** [reply]
Are the Framework XML/JSON, XSD, Regex still relevant now as they can be done in a Logic Apps Standard environment now with built-in functions (w/o Integration Account)

**Reply by pim-simons** [reply]
Regex is used quite a lot through the exception handler, the other ones I am not sure about, my earlier comment is still relevant I think:

>  @GoutsmitSam @Pauwelz we should checkout which projects this component is used and what the usecase for it is.
